Dockerfile

```diff
@@ -1,71 +1,50 @@
-# Build compilation image
-FROM ruby:3.1.4-alpine as builder
+FROM ruby:3.1.4-alpine3.18 as base
 
-# The application runs from /app
 WORKDIR /app
 
-# Add the timezone as it's not configured by default in Alpine
+# Add the timezone as it’s not configured by default in Alpine
 RUN apk add --update --no-cache tzdata && \
   cp /usr/share/zoneinfo/Europe/London /etc/localtime && \
   echo "Europe/London" > /etc/timezone
 
-# build-base: complication tools for bundle
+# build-base: compilation tools for bundle
 # yarn: node package manager
 # postgresql-dev: postgres driver and libraries
-RUN apk add --no-cache build-base yarn postgresql-dev git
+RUN apk add --no-cache build-base yarn postgresql-dev git bash
 
-# Install bundler to run bundle exec
-# This should be the same version as the Gemfile.lock
+# Bundler version should be the same version as what the Gemfile.lock was bundled with
 RUN gem install bundler:2.3.14 --no-document
 
-# Install gems defined in Gemfile
 COPY .ruby-version Gemfile Gemfile.lock /app/
+RUN bundle install --jobs=4 --no-binstubs --no-cache
 
-RUN bundle config set without "development test"
-ARG BUNDLE_FLAGS="--jobs=4 --no-binstubs --no-cache"
-RUN bundle install ${BUNDLE_FLAGS}
-
-# Install node packages defined in package.json, including webpack
 COPY package.json yarn.lock /app/
 RUN yarn install --frozen-lockfile
 
-# Copy all files to /app (except what is defined in .dockerignore)
 COPY . /app/
 
-# Compile assets and run webpack. We set a dummy secret key.
-RUN RAILS_ENV=production bundle exec rails SECRET_KEY_BASE=pickasecuretoken assets:precompile
+RUN bundle exec rake assets:precompile
 
-# Cleanup to save space in the production image
-RUN rm -rf node_modules log tmp && \
```

Comment:
We don't know why the node modules folder was being deleted in production when it was being created by the `yarn install` step. We're of the opinion that it would need it in order to run?

As for the other files, we're unsure how appropriate they are and whether it's having much benefit removing them, so opted to leave out for now, and observe image sizes / image build speed in the future

Reply:
Without looking in detail, there is at least one asset compilation step above - I'd expect that to create everything the app is actually using and render node_modules etc unnecessary. In general that's a fairly common pattern - you need various dependencies around to build from but then you only need your built/compiled artifacts to actually run, and making sure you have as little as possible on your production image is ideal.

It would be worth looking into that/what's going on here a bit more.

Reply:
So it looks like these are ok files to delete in the end. We have implemented this for staging and production - env's where we'd want to keep the image size low. 

We're also deleting the `frontend` folder because we think it's redundant after compiling the assets

However we are no longer deleting the `log` and `tmp` folders, as we discovered that the app tries to make these when it starts up (and needs permissions to do as well)